### PR TITLE
First iteration of relational API for R client

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

File: tools/rpkg/tests/testthat/test_relational.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    # yay R one-liners
+    gen_rand_string <- function(x, max_len) paste0(chars[sample(1:length(chars), runif(1)*max_len, replace=TRUE)], collapse="")
+
+    test_string_vec <- c(vapply(1:n, gen_rand_string, "character", max_len), NA, NA, NA, NA, NA, NA, NA, NA) # batman
+
+    df <- data.frame(s=test_string_vec, stringsAsFactors=FALSE)
+    expect_equivalent(df, as.data.frame(duckdb::rel_from_df(con, df)))
+
+    res <- duckdb::rel_from_df(con, df) |> duckdb::rel_sql("SELECT s::string FROM _")
```
Comment (Mytherin) [inline tools/rpkg/tests/testthat/test_relational.R:74]: What if we try to create a (persistent) table from a DF holding these strings? Or is that not possible (yet) with the current setup?

Comment (hannesmuehleisen) [reply tools/rpkg/tests/testthat/test_relational.R:74]: Its not possible yet

File: tools/rpkg/NAMESPACE [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Generated by roxygen2: do not edit by hand
 
-S3method(print,duckdb_explain)
```
Comment (rsund) [inline tools/rpkg/NAMESPACE:3]: No need to remove this, please re-run `roxygen` before merge.

File: tools/rpkg/NAMESPACE [hunk 3]
```diff
@@ -11,10 +15,20 @@ export(duckdb_register_arrow)
...
+export(rel_aggregate)
+export(rel_explain)
+export(rel_filter)
+export(rel_from_df)
+export(rel_order)
+export(rel_project)
+export(rel_sql)
 exportClasses(duckdb_connection)
 exportClasses(duckdb_driver)
-exportClasses(duckdb_explain)
```
Comment (rsund) [inline tools/rpkg/NAMESPACE:17]: No need to remove this, please re-run `roxygen` before merge.

Comment (hannesmuehleisen) [reply tools/rpkg/NAMESPACE]: Thanks for spotting, no longer removed. Not sure what happened there.

**Pull request closed without merging**